assistant/assistant.go

```diff
@@ -241,20 +241,28 @@ func (assistant *Assistant) getPlugin(req *request) response {
 
 type schema struct {
 	Types    map[string]interface{}
-	Defaults interface{}
+	Defaults map[string]interface{}
 }
 
 // getSchema returns the struct response containing config schema for a single plugin
 func (assistant *Assistant) getSchema(req *request) response {
 	fmt.Print("D! [assistant] Received request: ", req.Operation, " for plugin ", req.Plugin.Name, "\n")
 
 	var plugin interface{}
+	var defaultValues map[string]interface{}
 	var err error
+
 	switch req.Plugin.Type {
 	case "INPUT":
-		plugin, err = assistant.agent.GetDefaultInputPlugin(req.Plugin.Name)
+		plugin, err = assistant.agent.CreateInput(req.Plugin.Name)
+		defaultValues, err = assistant.agent.GetDefaultInputPlugin(req.Plugin.Name)
 	case "OUTPUT":
-		plugin, err = assistant.agent.GetDefaultOutputPlugin(req.Plugin.Name)
+		plugin, err = assistant.agent.CreateOutput(req.Plugin.Name)
+		defaultValues, err = assistant.agent.GetDefaultOutputPlugin(req.Plugin.Name)
```

Comment:
Are these plugin variables used anywhere else? Why do we need to create these plugin objects